apps/meteor/ee/client/views/admin/deviceManagement/DeviceInfoContextualBar.tsx

```diff
@@ -0,0 +1,135 @@
+import { DeviceManagementPopulatedSession } from '@rocket.chat/core-typings';
+import { Box, Button, StatusBullet } from '@rocket.chat/fuselage';
+import { useRoute, useTranslation, useToastMessageDispatch, useSetModal } from '@rocket.chat/ui-contexts';
+import React, { ReactElement, useCallback, useMemo } from 'react';
+
+import GenericModal from '../../../../../client/components/GenericModal';
+import VerticalBar from '../../../../../client/components/VerticalBar';
+import UserAvatar from '../../../../../client/components/avatar/UserAvatar';
+import { useEndpointAction } from '../../../../../client/hooks/useEndpointAction';
+import { useFormatDateAndTime } from '../../../../../client/hooks/useFormatDateAndTime';
+import { usePresence } from '../../../../../client/hooks/usePresence';
+import InfoPanel from '../../../../../client/views/InfoPanel';
+
+type DeviceInfoContextualBarProps = DeviceManagementPopulatedSession & {
+	onReload: () => void;
+};
+
+const DeviceInfoContextualBar = ({
+	device,
+	sessionId,
+	loginAt,
+	ip,
+	userId,
+	_user,
+	onReload,
+}: DeviceInfoContextualBarProps): ReactElement => {
+	const t = useTranslation();
+	const deviceManagementRouter = useRoute('device-management');
+	const formatDateAndTime = useFormatDateAndTime();
+	const dispatchToastMessage = useToastMessageDispatch();
+	const setModal = useSetModal();
+
+	const { name: clientName, os } = device || {};
+	const { username, name } = _user;
```

Comment:
if there's no user, it's going to break the page, we should prevent it even in the render